### ISSUE TITLE
Support starting inspector through signal

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#9312]
In some cases it is useful to be able to start the inspector on an already running process (for example a long running task that you can not easily restart). Deno currently has no infrastructure to support this. Currently the only way to enable the debugger is through `--inspect` or `--inspect-brk`.

In Node you can start the inspector on an already running process by sending the process a `SIGUSR1` signal. See Node docs: https://nodejs.org/api/process.html#process_signal_events & article on the topic: https://blog.sqreen.com/memory-leaks-nodejs-heroku/.

It would be cool if we supported something like to this.

**Comment by caspervonb** [comment]
Yeah been thinking this as-well; but with one addition, another signal to drop into an interactive session.

**Comment by JamesKyburz** [comment]
If this was added, then maybe add something similar to node's process._debugProcess to also support windows.

**Comment by bartlomieju** [comment]
Inspector instance is now created for all isolate, so technically there's nothing preventing us from implementing this feature (which would mean spawning a WS inspector server when signal is received).

**Comment by fserb** [comment]
Could we support this? Or at least a programatically way to spawn the server (`Deno.startInspectServer()` or something)?

**Comment by bartlomieju** [comment]
You can already activate the inspector on an already-running process programmatically via `node:inspector`:

```js
import inspector from "node:inspector";
inspector.open(9229, "127.0.0.1", false);
console.log(inspector.url()); // ws://127.0.0.1:9229/...
```

This prints `Debugger listening on ws://...` and is immediately attachable (verified, including `/json/list` returning a `webSocketDebuggerUrl`). If you want signal-triggered activation, you can opt in explicitly:

```js
import inspector from "node:inspector";
Deno.addSignalListener("SIGUSR1", () => {
  if (!inspector.url()) inspector.open();
});
```

What we are not going to add is Node's built-in, zero-config `SIGUSR1` handler that starts the inspector with no code. Installing a default signal handler on every process adds to startup cost for a feature that is now fully achievable opt-in via `inspector.open()` (optionally wired to a signal as above). Closing on that basis — the underlying need (debugging a long-running process without restarting it) is covered.

**Comment by tredondo** [comment]
I started a process weeks ago (without `--inspect`), and is now stuck (logging stopped). Is there a way to attach to it and debug it?

> You can already activate the inspector on an already-running process programmatically

Just to clarify, in my situation it's impossible to attach to that process.

**Comment by bartlomieju** [comment]
> Is there a way to attach to it and debug it?

No

> Just to clarify, in my situation it's impossible to attach to that process.

Correct
